tokio/src/io/uring/write.rs

```diff
@@ -29,27 +41,20 @@ impl Cancellable for Write {
 }
 
 impl Op<Write> {
-    /// Issue a write that starts at `buf_offset` within `buf` and writes some bytes
-    /// into `file` at `file_offset`.
-    pub(crate) fn write_at(
-        fd: OwnedFd,
-        buf: OwnedBuf,
-        buf_offset: usize,
-        file_offset: u64,
-    ) -> io::Result<Self> {
+    /// Issue a write at `file_offset` from the provided `buf`. To use current file cursor, set `file_offset` to `-1` or `u64::MAX`.
+    pub(crate) fn write_at(fd: ArcFd, buf: blocking::Buf, file_offset: u64) -> Self {
```

Comment:
This seems to be a feature that relies on a flag `IORING_FEAT_RW_CUR_POS` being present. We should probably make sure to check this feature somehow.